src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+public class Parser {
+    public static Command parse(String text, State state) throws DukeException, DateTimeParseException {
```

Comment:
Perhaps you could break down your parse method into smaller methods like parseAddCommand etc.?